aruco_ros/CMakeLists.txt

```diff
@@ -23,98 +21,112 @@ set(THIS_PACKAGE_INCLUDE_DEPENDS
     sensor_msgs
     visualization_msgs
 )
+
 find_package(ament_cmake REQUIRED)
-foreach(Dependency IN ITEMS ${THIS_PACKAGE_INCLUDE_DEPENDS})
-  find_package(${Dependency} REQUIRED)
+foreach(DEP ${dependencies})
+  find_package(${DEP} REQUIRED)
 endforeach()
 
-#generate_dynamic_reconfigure_options(
-#  cfg/ArucoThreshold.cfg
-#)
-
-add_library(aruco_ros_utils SHARED src/aruco_ros_utils.cpp)
-target_include_directories(aruco_ros_utils
+# Build library.
+add_library(${PROJECT_NAME}_utils SHARED
+  src/aruco_ros_utils.cpp
+)
+target_include_directories(${PROJECT_NAME}_utils
   PUBLIC
-  include)
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include/${PROJECT_NAME}>)
 
-target_include_directories(aruco_ros_utils
-  SYSTEM PUBLIC
+target_link_libraries(${PROJECT_NAME}_utils PUBLIC
+  ${OpenCV_LIBRARIES}
+  tf2_geometry_msgs::tf2_geometry_msgs
+  cv_bridge::cv_bridge
+)
+
+target_include_directories(${PROJECT_NAME}_utils PUBLIC
   ${OpenCV_INCLUDE_DIRS}
+  ${aruco_INCLUDE_DIRS}
+  ${sensor_msgs_INCLUDE_DIRS}
+  ${visualization_msgs_INCLUDE_DIRS}
 )
-ament_target_dependencies(aruco_ros_utils ${THIS_PACKAGE_INCLUDE_DEPENDS})
-target_link_libraries(aruco_ros_utils ${OpenCV_LIBRARIES})
 
+# Build executables.
 add_executable(single src/simple_single.cpp
                       src/aruco_ros_utils.cpp)
```

Comment:
This cpp file should not be there as it is compiled before in `${PROJECT_NAME}_utils` and linked to this executable.